src/schema/v2/me/delete_account_mutation.ts

```diff
@@ -0,0 +1,73 @@
+import { GraphQLBoolean, GraphQLObjectType, GraphQLUnionType } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { ResolverContext } from "types/graphql"
+
+const AccountMutationDeleteSuccessType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "AccountMutationSuccess",
+  isTypeOf: (data) => {
+    return data._type !== "GravityMutationError"
+  },
+  fields: () => ({
+    success: {
+      type: GraphQLBoolean,
+      resolve: () => true,
+    },
+  }),
+})
+
+const AccountMutationFailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "AccountMutationFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => (typeof err.message === "object" ? err.message : err),
+    },
+  }),
+})
+
+export const AccountMutationType = new GraphQLUnionType({
+  name: "AccountMutationType",
+  types: [AccountMutationDeleteSuccessType, AccountMutationFailureType],
+})
+
+export const deleteUserAccountMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "DeleteAccount",
+  description: "Delete User Artsy Account",
+  outputFields: {
+    userAccountOrError: {
+      type: AccountMutationType,
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: ({ deleteUserAccountLoader }) => {
+    if (!deleteUserAccountLoader) {
+      throw new Error("You need to be signed in to perform this action")
```

Comment:
@dzucconi we're hitting this throw line when passing an `x-access-token` header - is that what you'd expect? 🤔